vars/downloadFromS3.groovy

```diff
@@ -6,10 +6,21 @@
  * this file be licensed under the Apache-2.0 license or a
  * compatible open source license.
  */
+ /**
+Library to download the artifacts from S3 bucket to local
+@param Map[assumedRoleName] <Required> - Role to be assumed to download artifacts.
+@param Map[roleAccountNumberCred] <Required> - AWS account number to download artifacts from.
+@param Map[bucketName] <Required> -S3 bucket Name.
+@param Map[downloadPath] <Required> - This is the path inside the bucket to use.
+@param Map[localPath] <Required> - This is the local target file to download into.
+@param Map[force]<Optional> - Set this to true to overwrite local workspace files. Defaults to false
+*/
 void call(Map args = [:]) {
-    withCredentials([string(credentialsId: 'jenkins-aws-account-public', variable: 'AWS_ACCOUNT_PUBLIC')]) {
-            withAWS(role: 'opensearch-bundle', roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName: 'jenkins-session') {
-                s3Download(file: args.destPath, bucket: args.bucket, path: args.path, force: args.force)
+    boolean forceDownload = args.force ?: false
+
+    withCredentials([string(credentialsId: "${args.roleAccountNumberCred}", variable: 'AWS_ACCOUNT_NUMBER')]) {
```

Comment:
Should we ask team to put specific account number as credentialID here? This would expose the AWS account number right?

Reply:
Its jenkins credential id so AWS secrets manager name. ex: `jenkins-aws-account-number` which in AWS secrets manager will hold the actual account number

Reply:
See: https://github.com/opensearch-project/opensearch-build-libraries/pull/135/files#diff-1276b5e664e1a19d973a05e19d7020a26a4ddfaa1bf025c723bc988ce65f2125R18

Reply:
Oh so we would need to store each of the account we would download from to the secret manager right?

Reply:
Yes, thats the plan